test/Base_Tests/src/Data/Decimal_Spec.enso

```diff
@@ -619,6 +628,24 @@ add_specs suite_builder =
             f.should_be_a Float
             f.should_equal 56.34
 
+        group_builder.specify "Conversion from float to decimal should use the correct constructor" <|
+            is_value_constructor d = case d of
+                Decimal.Value _ -> True
+                Decimal.From_Float _ _ -> False
+
+            is_value_constructor (0.1 . to Decimal) . should_be_true
+            is_value_constructor (Decimal.from 0.1) . should_be_false
+            is_value_constructor (Decimal.from 0.1 explicit=True) . should_be_true
+            is_value_constructor (id_decimal 0.1) . should_be_false
+
+        group_builder.specify "Arithmetic with decimal and explicitly converted float should succeed" <|
+            ((Decimal.new 1) + (0.1 . to Decimal)) . should_equal 1.1
+            ((Decimal.new 1) + (Decimal.from 0.1 explicit=True)) . should_equal 1.1
+
+        group_builder.specify "Arithmetic with decimal and implicitly converted float should fail" <|
+            ((Decimal.new 1) + (Decimal.from 0.1)) . should_fail_with Illegal_Argument
+            ((Decimal.new 1) + (id_decimal 0.1)) . should_fail_with Illegal_Argument
```

Comment:
Do we have some explanation anywhere on how these 2 differ and why?

I always saw `to` as just a 'reversed' variant of `from`, I don't see how one is more explicit than the other.

`Decimal.from f` is as explicit as `f.to Decimal`; only `(f : Decimal)` could be called implicit I guess.

I'm not against the changes but I'm confused by them - as a user I don't see a difference between `to` and `from` and thus I'm surprised by differing behaviour, yet I did not notice (did I miss something?) any explanation of this difference.

Reply:
See answer to the other comment.